pkg/commands/os.go

```diff
@@ -57,6 +58,53 @@ func (c *OSCommand) RunCommandWithOutput(command string) (string, error) {
 	)
 }
 
+// RunCommandWithOutputLive runs RunCommandWithOutputLiveWrapper
+func (c *OSCommand) RunCommandWithOutputLive(command string, output func(string) string) (errorMessage string, err error) {
+	return RunCommandWithOutputLiveWrapper(c, command, output)
+}
+
+// DetectUnamePass detect a username / password question in a command
+// ask is a function that gets executen when this function detect you need to fillin a password
+// The ask argument will be "username" or "password" and expects the user's password or username back
+func (c *OSCommand) DetectUnamePass(command string, ask func(string) string) error {
+	ttyText := ""
+	errMessage, err := c.RunCommandWithOutputLive(command, func(word string) string {
+		ttyText = ttyText + " " + word
+
+		type Prompt struct {
+			pattern   string
+			canAskFor bool
+		}
+		prompts := map[string]Prompt{
+			"password": {
+				pattern:   `Password\s*for\s*'.+':`,
+				canAskFor: true,
+			},
+			"username": {
+				pattern:   `Username\s*for\s*'.+':`,
+				canAskFor: true,
+			},
+		}
+
+		for askFor, prompt := range prompts {
+			if match, _ := regexp.MatchString(prompt.pattern, ttyText); match && prompt.canAskFor {
```

Comment:
```suggestion
			if !prompt.canAskFor {
				continue
			}
			if match, _ := regexp.MatchString(prompt.pattern, ttyText); match {
```

Reply:
I'm also curious as to whether we need the canAskFor flag. Given that we're setting ttyText = "", doesn't that mean that after entering a username, we're not going to see another prompt for a username in ttyText?

Reply:
This makes sure to only ask once per command for a username / password.  
Git does only ask for the username/password once per git pull,push and fetch.
Also because this function will get called per command the `canAskFor` doesn't effect other git pulls, pushes and fetches (it will ask every time for a username and password if needed).  

Beside that the reason why that is there because if i created a file named `Password for 'test':` and make sure that it is visible in the git pull message lazygit will break but this case is very unlikely because who would write a file named like that.  
![image](https://user-images.githubusercontent.com/15320763/49696226-445f3300-fba7-11e8-9f2c-269cb82e03e1.png)